pmm/pmm2-upgrade-portal-tests.groovy

```diff
@@ -0,0 +1,277 @@
+            name: 'PMM_SERVER_VERSION')
+        choice(
+            choices: ['2.27.0', '2.26.0', '2.25.0'],
+            description: "PMM Client Version",
+            name: 'PMM_CLIENT_VERSION')
+        choice(
+            choices: ['Production', 'Testing', 'Experimental'],
+            description: "Select Repository for testing",
+            name: 'PMM_REPOSITORY')
+        string(
+            defaultValue: 'admin-password',
+            description: 'Change pmm-server admin user default password.',
+            name: 'ADMIN_PASSWORD')
+        string(
+            defaultValue: 'main',
+            description: 'Tag/Branch for pmm-qa repository',
+            name: 'PMM_QA_GIT_BRANCH')
+    }
+    options {
+        skipDefaultCheckout()
+    }
+    stages {
+        stage('Prepare') {
+            steps {
+                // clean up workspace and fetch pmm-ui-tests repository
+                deleteDir()
+                git poll: false, branch: GIT_BRANCH, url: 'https://github.com/percona/pmm-ui-tests.git'
+                wrap([$class: 'BuildUser']) {
+                    sh """
+                        echo "\${BUILD_USER_EMAIL}" > OWNER_EMAIL
+                        echo "\${BUILD_USER_EMAIL}" | awk -F '@' '{print \$1}' > OWNER_FULL
+                    """
+                }
+                script {
+                    slackSend botUser: true, channel: '#pmm-ci', color: '#FFFF00', message: "[${JOB_NAME}]: build started - ${BUILD_URL}"
+                }
+                sh '''
+                    sudo yum -y update --security
+                    sudo yum -y install php php-mysqlnd php-pdo jq svn bats mysql
+                    curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+                    sudo install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
+                    sudo amazon-linux-extras install epel -y
+                    sudo mkdir -p /srv/pmm-qa || :
+                    pushd /srv/pmm-qa
+                        sudo git clone --single-branch --branch \${PMM_QA_GIT_BRANCH} https://github.com/percona/pmm-qa.git .
+                        sudo git checkout \${PMM_QA_GIT_COMMIT_HASH}
+                        sudo chmod 755 pmm-tests/install-google-chrome.sh
+                        bash ./pmm-tests/install-google-chrome.sh
+                    popd
+                    sudo ln -s /usr/bin/google-chrome-stable /usr/bin/chromium
+                '''
+            }
+        }
+        stage('Checkout Commit') {
+            when {
+                expression { env.GIT_COMMIT_HASH.length()>0 }
+            }
+            steps {
+                sh 'git checkout ' + env.GIT_COMMIT_HASH
+            }
+        }
+        stage('Start Server') {
+            steps {
+                runStagingServer("perconalab/pmm-server:$PMM_SERVER_VERSION", "$PMM_CLIENT_VERSION" , "no", '127.0.0.1', ADMIN_PASSWORD)
```

Comment:
What if I want to use an image from `percona/pmm-server` repo?

Reply:
change to string value, so user can select which repo he wants.